int32 pin
int32 val
---
int32 out